include/api.php

```diff
@@ -57,47 +67,74 @@ function pll_current_language( $field = 'slug' ) {
  * @since 1.0
  *
  * @param string $field Optional, the language field to return ( @see PLL_Language ), defaults to 'slug'. Pass OBJECT constant to get the language object.
- * @return string|PLL_Language|false The requested field for the default language.
+ * @return string|PLL_Language|false The requested field or object for the default language. False if none.
  */
 function pll_default_language( $field = 'slug' ) {
-	if ( isset( PLL()->options['default_lang'] ) ) {
-		$lang = PLL()->model->get_language( PLL()->options['default_lang'] );
-		if ( $lang ) {
-			if ( OBJECT === $field ) {
-				return $lang;
-			}
-			return isset( $lang->$field ) ? $lang->$field : false;
-		}
+	if ( empty( PLL()->options['default_lang'] ) ) {
+		return false;
+	}
+
+	$lang = PLL()->model->get_language( PLL()->options['default_lang'] );
+
+	if ( empty( $lang ) ) {
+		return false;
 	}
-	return false;
+
+	if ( OBJECT === $field ) {
+		return $lang;
+	}
+
+	return isset( $lang->$field ) ? $lang->$field : false;
 }
 
 /**
- * Among the post and its translations, returns the id of the post which is in the language represented by $lang.
+ * Among the post and its translations, returns the ID of the post which is in the language represented by $lang.
  *
  * @api
  * @since 0.5
+ * @since 3.4 Returns 0 instead of false.
+ * @since 3.4 $lang accepts PLL_Language or string.
+ *
+ * @param int                 $post_id Post ID.
+ * @param PLL_Language|string $lang    Optional language (object or slug), defaults to the current language.
+ * @return int|null The translation post ID if exists, otherwise the passed ID. `0` if the passed object has no language.
+ *                  `null` if the language is not defined yet.
  *
- * @param int    $post_id Post id.
- * @param string $lang    Optional language code, defaults to the current language.
- * @return int|false|null Post id of the translation if it exists, false otherwise, null if the current language is not defined yet.
+ * @phpstan-return int<0, max>|null
```

Comment:
Should we really distinguish `0` from `null`?

Reply:
Removed in e655b45